tests/integration/test_trace_log_table/test.py

```diff
@@ -0,0 +1,66 @@
+# pylint: disable=unused-argument
+# pylint: disable=redefined-outer-name
+import time
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+
+node = cluster.add_instance('node', with_zookeeper=True, main_configs=["configs/trace_log.xml"])
+
+
+@pytest.fixture(scope='module')
+def start_cluster():
+    try:
+        cluster.start()
+
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+# Tests that the event_time_microseconds field in the system.trace_log table gets populated.
+# To make the tests work better, the default flush_interval_milliseconds is being overridden
+# to 1000 ms. Also the query_profiler_real_time_period_ns and the query_profiler_cpu_time_period_ns
+# are set to suitable values so that traces are properly populated. Test compares the event_time and
+# event_time_microseconds fields and asserts that they are exactly equal upto their seconds parts. Also
+# one additional test to ensure that the count(event_time_microseconds) is > 0;
+@pytest.mark.skip(reason="TODO: system.trace_log not populated in time on CI but works fine on dev")
```

Comment:
I can't seem to get the `system.trace_log` to work on CI. This test runs just fine locally. On CI, it seems like the table exists, but there isn't any log in it yet. So. when the test runs, the asserts fail - I've marked to skip this for now. Is there another setting that need to be tweaked in order to get this to work?

Reply:
Sure, I will change it to an functional stateless test.

Reply:
@tavplubix  made this a functional test, but I've been fighting hard to get the tests for trace_log to pass. Please refer to my latest commit b08b089 and even with a long running query and ample amount of sleep, tests fail on CI.

Reply:
@tavplubix I need some help with this particular test which is has been tricky and haven't been successful to make this pass. . I will really appreciate some suggestions / alternate ways to fix this.